### PR TITLE
Feat/dapp staking v2 registration

### PR DESCRIPTION
**Pull request by bobo-k2**
**Pull Request Summary**

Since there we don't have register button yet, use https://astar-apps--pr522-feat-dapp-staking-v2-c3en1g09.web.app/#/astar/dapp-staking/register to access dapp registration page

This PR adds dapp staking registration page, layout, components and styling
Dapp submiting logic is not part of this PR

**Check list**
- [ ] contains breaking changes
- [x] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Adds**
- Dapps staking registration page.

**Fixes**
- (ex: Fix validation function)

**Changes**
- (ex: Change document B)

**To-dos**
> *Feel free to remove this section if it's not applicable

- [ ] (ex: add user list)

### CODE REVIEW

**Comment by Kahonnohak** [comment]
Thanks @bobo-k2 for a sync!

Please check the notes here.

Part A
https://drive.google.com/file/d/1RFb3fCac4QcP2HRzG2sTGdYRiAGAnA7G/view?usp=sharing
comments are in the link
Screenshot 2022-09-28 at 16 39 25


Part B
https://drive.google.com/file/d/1OdI1PDdaHEuIxiJeIYv9DaGu6wRNCFCq/view?usp=sharing
Screenshot 2022-09-28 at 16 38 46

Other than that, I will update the top area (congrats message)

Thanks

**Review by hoonsubin — APPROVED** [review]

File: src/components/dapp-staking/RegisterDapp.vue [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+      name: '',
+      iconFile: '',
+      linkedInAccountUrl: '',
+      twitterAccountUrl: '',
+    });
+
+    const { t } = useI18n();
+    const store = useStore();
+    const theme = computed<string>(() => store.getters['general/theme']);
+    const isDark = ref<boolean>(theme.value.toLowerCase() === 'dark');
```
Comment (impelcrypto) [inline src/components/dapp-staking/RegisterDapp.vue:132]: You can refer to global state.

```
const isDarkTheme = computed<boolean>(() => store.getters['general/theme'] === 'DARK');
```

File: src/components/dapp-staking/RegisterDapp.vue [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+      });
+    };
+
+    watch([theme], (val) => {
+      isDark.value = val[0].toLowerCase() === 'dark';
+    });
+
+    return {
+      data,
+      isDark,
```
Comment (impelcrypto) [inline src/components/dapp-staking/RegisterDapp.vue:213]: `isDark` is not used in the `template` in this file

**Review by impelcrypto — CHANGES_REQUESTED** [review]

File: src/components/dapp-staking/RegisterDapp.vue (hunk 2 above)
Comment (bobo-k2) [reply src/components/dapp-staking/RegisterDapp.vue]: > `isDark` is not used in the `template` in this file

Removed as not needed in [this](https://github.com/AstarNetwork/astar-apps/pull/522/commits/b5d06c46c916a0a8ba47c68485e871194e612ee2) commit

**Review by impelcrypto — APPROVED** [review]

File: src/components/dapp-staking/RegisterDapp.vue [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+    const currentCategory = ref<LabelValuePair>(possibleCategories[0]);
+    const dappForm = ref();
+    const errors = ref({
+      builders: '',
+      community: '',
+      platform: '',
+    });
+
+    // make a placeholder for add logo
+    data.icon = new File([], 'Add a logo image'); // TODO translate
```
Comment (sirius651) [inline src/components/dapp-staking/RegisterDapp.vue:141]: why don't we make these as property in json?

**Review by sirius651 — APPROVED** [review]

**Pull request merged**